clang/include/clang/Basic/Attr.td

```diff
@@ -1590,6 +1590,13 @@ def SYCLDevice : InheritableAttr {
   let Documentation = [SYCLDeviceDocs];
 }
 
+def SYCLDeviceOnly : InheritableAttr {
+  let Spellings = [GNU<"sycl_device_only">];
+  let Subjects = SubjectList<[Function]>;
+  let LangOpts = [SYCLIsDevice];
```

Comment:
```suggestion
  let LangOpts = [SYCLIsHost, SYCLIsDevice];
```

otherwise this would create a warning during host compilation and the filtering is dead code